packages/lit-dev-content/site/docs/components/context.md

```diff
@@ -0,0 +1,497 @@
+
+#### ContextConsumer
+
+ContextConsumer is a reactive controller that manages dispatching the `context-request` event for you. The controller will cause the host element to update when new values are provided. The provided value is then available at the `.value` property of the controller.
+
+```ts
+import {LitElement, property} from 'lit';
+import {ContextConsumer} from '@lit-labs/context';
+import {Logger, loggerContext} from './logger.js';
+
+export class MyElement extends LitElement {
+  private _myData = new ContextConsumer(this, myContext);
+
+  render() {
+    const myData = this._myData.value;
+    return html`...`;
+  }
+}
+```
+
+#### Subscribing to contexts
+
+Consumers can subscribe to context values so that if a provider has a new value, it can give it to all subscribed consumers, causing them to update.
+
+You can subscribe with the `@consume()` decorator:
+
+```ts
+  @consume({context: myContext, subscribe: true})
+  myData: MyData;
+```
+
+and the ContextConsumer controller:
+
+```ts
+  private _myData = new ContextConsumer(this,
+    myContext,
+    undedined, /* callback */
+    true /* subscribe */
+  );
+```
+
+## Example Use Cases
+
+### Current user, locale, etc.
+
+The most common context use cases involve data that is global to a page and possibly only sparsely needed in components throughout the page. Without context it's possible that most or all components would need to accept and propagate reactive properties for the data.
+
+### Services
+
+App-global services, like loggers, analytics, data stores, can be provided by context. An advantage of context over importing from a common module are the late coupling and tree-scoping that context provides. Tests can easily provide mock services, or different parts of the page can be given different service instances.
+
+### Themes
+
+Themes are sets of styles that apply to the entire page or entire subtrees within the page - exactly the kind of scope of data that context provides.
+
+One way of building a theme system would be to define a `Theme` type that containers can provide that holds named styles. Elements that want to apply a theme can consume the theme object and look up styles by name. Custom theme reactive controllers can wrap ContextProvider and ContextConsumer to reduce boilerplate.
+
+### HTML-based plugins
+
+Context can be used to provide data to light DOM children, which typically aren't created from the same declarative templates with databinding available.
+
+For example, consider a code editor element with plugins for different language modes. You can make a plain HTML system for adding features using context:
+
+```html
+<code-editor>
+  <code-editor-javascript-mode></code-editor-javascript-mode>
+  <code-editor-python-mode></code-editor-python-mode>
+</code-editor>
+```
+
+In this case `<code-editor>` would provide an API for adding langage modes via context, and plugin elements would consume that API and add themselves to the editor.
```

Comment:
I don't super love this example; it's definitely valid, but could we choose one where both are visual (rather than the headless/utility element example)?

Reply:
I think because we're not showing anything visual, that this might be easier to understand.

It is a case a came across a few times actually, where an editor maintainer said they didn't see the value in custom elements because you had to configure the editor in JS anyway.